src/frontend/src/pages/view-challenge/components/popup.tsx

```diff
@@ -0,0 +1,19 @@
+import React from "react";
+import { UnkownChallenge } from "../../../types";
+import CorrectAnswerPopup from "./correct-answer-popup";
+import IncorrectAnswerPopup from "./incorrect-answer-popup";
+
+interface PopupProps {
+    challenge: UnkownChallenge;
+    selected: string;
+}
+
+const Popup: React.FC<PopupProps> = ({ challenge, selected }) => {
```

Comment:
```suggestion
const Popup: React.FC<PopupProps> = (props: PopupProps) => {
```
The general rule in React is that you should destructure as late as possible. props should be accessed by props.whatever instead of destructuring in the arguments like this.